src/LuaMultiThreaded2.cpp

```diff
@@ -68,6 +69,47 @@ extern "C" int errorHandler(LuaState * L)
 
 
 
+extern "C" static int mutexNew(LuaState * aState)
+{
+	lua_pushcfunction(aState, errorHandler);
+
+	// Push the (currently empty) mutex object to the Lua side
+	auto mutexObj = reinterpret_cast<std::mutex**>(lua_newuserdata(aState, sizeof(std::mutex**)));
+	luaL_setmetatable(aState, MUTEX_METATABLE_NAME);
+
+	// Create the new mutex:
+	*mutexObj = new std::mutex();
+	return 1;
+}
+
+
+
+
+
+
+
+/** */
+extern "C" static int mutexLock(LuaState * aState)
+{
+	auto mutexObj = reinterpret_cast<std::mutex**>(luaL_checkudata(aState, 1, MUTEX_METATABLE_NAME));
+	if (mutexObj == nullptr)
+	{
+		luaL_argerror(aState, 0, "'mutex' expected");
+		return 0;
+	}
+	(*mutexObj)->lock();
+	auto numParams = lua_gettop(aState);
+	luaL_checktype(aState, 2, LUA_TFUNCTION);
+	lua_pcall(aState, 0, 0, 0);
```

Comment:
Perhaps also allow passing any params? This way it doesn't need to go through the registry, just copy-`push` them onto the stack.